core/controllers/story_viewer.py

```diff
@@ -170,6 +171,7 @@ def get(self, story_id, node_id):
     @acl_decorators.can_access_story_viewer_page
     def post(self, story_id, node_id):
         story = story_fetchers.get_story_by_id(story_id)
```

Comment:
Handle deleted story here as well. Ditto elsewhere where you fetch any model, handle the case where the model is deleted.